### PR TITLE
feat: add demo slides for using beamer lectures

### PR DESCRIPTION
**Pull request by Benno2000**
I created some demo slides to illustrate how to use beamer lectures with the template.

### CODE REVIEW

**Comment by tthuem** [comment]
I guess we need at least three lecture parts to illustrate how it works (e.g., with includeonlylecture). Also, maketitle should be used for every lecture, ideally also with a different lecture title (e.g., subtitle).

File: lecture-demo/lectureDemo.tex [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+\title{Beamer Lecture Test}
+\author{Author}
+
+\includeonlylecture{lecture 1}
+
+\begin{document}
+    \maketitle
+
+    %%% Lecture 1 %%%
```
Comment (tthuem) [inline lecture-demo/lectureDemo.tex:16]: overview on lectures still missing

File: lecture-demo/lectureDemo.tex [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    \section{First Section of Lecture \thelecture}
+
+    \begin{frame}{First Slide of Lecture \thelecture}
+        Hello World! This section \thesection\ of \insertlecture.
+    \end{frame}
+
+    \section{Second Section of Lecture \thelecture}
+
+    \begin{frame}{First Slide of Lecture \thelecture}
```
Comment (tthuem) [inline lecture-demo/lectureDemo.tex:32]: Second Slide?

Why not using subsections here?

File: lecture-demo/lectureDemo.tex (hunk 1 above)
Comment (tthuem) [reply lecture-demo/lectureDemo.tex:16]: also it would be good to illustrate how jumps between lectures work

File: lecture-demo/lectureDemo.tex [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	aspectratio=169, % default is 43
+	8pt, % font size, default is 11pt
+]{beamer}
+
+\usepackage{../beamerthemeuulm} % use the inofficial uulm beamer theme
+
+\title{Beamer Lecture Test}
+\author{Author}
+
+\includeonlylecture{lecture 1}
```
Comment (tthuem) [inline lecture-demo/lectureDemo.tex:11]: would be good to indicate also how to generate the second or all lectures (e.g., with commands in comments)

File: beamerthemeuulm.sty [hunk 4]
```diff
@@ -262,6 +262,15 @@
 % ---------------------------------------------
 
 
+% ---------------------------------------------
+% handling lectures
+% ---------------------------------------------
+\AtBeginLecture{
+    \part{\insertlecture}\label{lec:\thelecture}
+}
```
Comment (tthuem) [inline beamerthemeuulm.sty:270]: I am missing subtitles at the beginning of each lecture (could indicate the topic of each lecture)

File: lecture-demo/lectureDemo.tex [hunk 5]
```diff
@@ -0,0 +1,78 @@
+\documentclass[
+	aspectratio=169, % default is 43
+	8pt, % font size, default is 11pt
+]{beamer}
+
+\usepackage{../beamerthemeuulm} % use the inofficial uulm beamer theme
+
+\title{Beamer Lecture Test}
```
Comment (tthuem) [inline lecture-demo/lectureDemo.tex:8]: would be good to use a caption here such that it is clear whether this is the lecture series or a single 90-minute lecture

**Review by tthuem — CHANGES_REQUESTED** [review]
We are on a good way, but still some parts are missing.

**Comment by Benno2000** [comment]
I updated the example slides for using lectures with our template:
- The `lectureDemo`-slides define a `\lectureoverviewslide` command that can once be manually adjusted to contain all lectures. It then displays the **list of all defined lectures** on the left side and automatically generates a **toc of the current lecture** on the right side.
- There are **three lectures** with **two sections** each to illustrate how lectures can be used together with sections.
- At the beginning of each lecture, a **titleslide** and a **lecture overview** is generated.
- It is illustrated how only **single lectures** can be included using `\includeonlylecture`

Regarding the template, there are a few modifications in `\AtBeginLecture`:
- At the beginning of each lecture, a new `\part` is being generated in order to **reset the sections** for the toc.
- The **section** and **frame numbers** are reset to `0`
- The **subtitle** is set to the **current lecture number** and **title**. This enables generating a new title slide for each lecture that has the overall title as title and the lecture title underneath it as subtitle.

**Comment by tthuem** [comment]
@Benno2000 The links in the PDF to other lectures do not work. Is it a problem on my end? Does it work on your setup?

**Comment by tthuem** [comment]
The overview slide still has a weird footer, but we can also fix that later:
image

**Comment by Benno2000** [comment]
@tthuem 

> The links in the PDF to other lectures do not work. Is it a problem on my end? Does it work on your setup?

When I tried it a few days ago, the links were not clickable within VS-Code but in Adobe Acrobat it worked fine. However it that does not work anymore now, too. I'll try to figure out what the issue is here.

> The overview slide still has a weird footer, but we can also fix that later

That's right, I'd suggest to fix that in a separate issue. For now, we could just use `[plain]` for the overview slides, which looks better in my opinion anyway in this case.

**Comment by Benno2000** [comment]
I fixed the lecture links by using `\hyperlink` instead of `\href`. Now they should be clickable.

**Comment by tthuem** [comment]
@Benno2000  Using [plain] for the overview slides sounds good, have not thought of that yet. Go ahead with the merge then.

**Pull request merged**